zitadel/v2/helper/client.go

```diff
@@ -44,28 +44,32 @@ func GetClientInfo(insecure bool, domain string, token string, jwtProfileFile st
 		return nil, fmt.Errorf("either 'jwt_profile_file' or 'jwt_profile_json' is required")
 	}
 
-	issuer := ""
-	if port != "" {
-		domain = domain + ":" + port
-		issuer = domain
-	} else {
-		issuer = domain
-		if insecure {
-			domain = domain + ":80"
-		} else {
-			domain = domain + ":443"
-		}
-	}
-
+	issuerScheme := "https://"
 	if insecure {
 		options = append(options, zitadel.WithInsecure())
-		issuer = "http://" + issuer
-	} else {
-		issuer = "https://" + issuer
+		issuerScheme = "http://"
+	}
+
+	issuerPort := port
+	if port == "80" && insecure || port == "443" && !insecure {
+		issuerPort = ""
+	}
+
+	issuer := issuerScheme + domain + ":" + issuerPort
+	if issuerPort == "" {
+		issuer = issuerScheme + domain
+	}
```

Comment:
Wouldn't it be easier to understand if we append the port if it's set?
(Sorry im on the phone)
Issier= scheme+domain
if port != "" 
Issuer += ":"+ port

Reply:
Same for the lines below

Reply:
Here yes. The lines below, no, because we append the default port to the clientDomain if the port argument is empty.

Pushed the change.